release/ray_release/cluster_manager/minimal.py

```diff
@@ -248,6 +251,39 @@ def create_cluster_compute(self, _repeat: bool = True):
                     f"ID {self.cluster_compute_id}"
                 )
 
+    @property
+    def cluster_compute_with_extra_tags(self) -> dict:
+        if self.extra_tags and self._get_cloud_provider() == "AWS":
+            cluster_compute = self.cluster_compute.copy()
+            if "aws" in cluster_compute:
+                cluster_compute["aws"] = deepcopy(cluster_compute["aws"])
+            else:
+                cluster_compute["aws"] = {}
+            cluster_compute["aws"].setdefault("TagSpecifications", [])
+            tag_specifications = cluster_compute["aws"]["TagSpecifications"]
+            for resource in self.extra_tags_resource_types:
+                resource_tags: dict = next(
+                    (
+                        x
+                        for x in tag_specifications
+                        if x.get("ResourceType", "") == resource
+                    ),
+                    None,
+                )
+                if resource_tags is None:
+                    resource_tags = {"ResourceType": resource, "Tags": []}
+                    tag_specifications.append(resource_tags)
+                tags = resource_tags["Tags"]
+                for key, value in self.extra_tags.items():
+                    tags.append({"Key": key, "Value": value})
+            return cluster_compute
+        else:
+            return self.cluster_compute
+
+    def _get_cloud_provider(self) -> str:
+        assert self.cluster_compute and "cloud_id" in self.cluster_compute
+        return self.sdk.get_cloud(self.cluster_compute["cloud_id"]).result.provider
```

Comment:
We should probably try/except this

Reply:
not sure what the behavior should be if the assertion is failed? besides, cloud id is required anyway, right